ldp/nn/handlers/transformer_handler.py

```diff
@@ -0,0 +1,1017 @@
+from __future__ import annotations
+
+import logging
+import os
+import socket
+import sys
+from abc import ABC, abstractmethod
+from collections.abc import Awaitable, Callable
+from enum import StrEnum, auto
+from functools import cache, partial, wraps
+from pathlib import Path
+from typing import Any, Concatenate, ParamSpec, Self, TypeVar, cast
+
+import accelerate
+import torch
+import torch.distributed as dist
+import tree
+from dask import config
+from dask.distributed import Client
+from dask_jobqueue import SLURMCluster
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+from torch import nn
+from torch.cuda import nccl
+from torch.distributed.fsdp import (
+    BackwardPrefetch,
+    FullStateDictConfig,
+    FullyShardedDataParallel,
+    MixedPrecision,
+    ShardingStrategy,
+    StateDictType,
+)
+from torch.nn.functional import pad
+from transformers import PreTrainedModel, PreTrainedTokenizer
+from transformers.generation.utils import GenerateDecoderOnlyOutput
+from transformers.tokenization_utils_base import BatchEncoding
+
+from ldp.graph.async_torch import AsyncBufferedWorker, AsyncTorchModule
+
+from ..lm_config import LMConfig, TorchDType  # noqa: TID252
+from ..utils import REPO_ROOT, set_seed  # noqa: TID252
+from .chunking import TensorChunker
+from .module_handler import ModuleExecutionInterface, ModuleHandler
+
+if sys.version_info >= (3, 12):
+    from typing import overload
+else:
+    from typing_extensions import overload  # noqa: UP035
+
+try:
+    from dask_cuda import LocalCUDACluster
+except ImportError:
+    LocalCUDACluster = None
+
+config.set({
```

Comment:
Given we are a package people are installing, shouldn't we not being doing module-level configuration? Can we move this to a configuration function that one can import/use?

Reply:
Yeah I see what you mean, but I'll think about this later. We're doing lots of env-level configuration, like setting CUDA default settings